server/util.go

```diff
@@ -276,31 +287,110 @@ func (rf *redirectFormatter) Format(pkg string, level capnslog.LogLevel, depth i
 // Flush only for implementing Formatter.
 func (rf *redirectFormatter) Flush() {}
 
+// modifyHook injects file name and line pos into log entry.
+type contextHook struct{}
+
+// Fire implements logrus.Hook interface
+// https://github.com/sirupsen/logrus/issues/63
+func (hook *contextHook) Fire(entry *log.Entry) error {
+	pc := make([]uintptr, 3, 3)
+	cnt := runtime.Callers(6, pc)
+
+	for i := 0; i < cnt; i++ {
+		fu := runtime.FuncForPC(pc[i] - 1)
+		name := fu.Name()
+		if !strings.Contains(name, "github.com/Sirupsen/logrus") &&
```

Comment:
for standalone sub-package, a function is ok.
for mess up with server/util.go, another function may not be acceptable.

never mind 😹